src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1306,63 +1315,60 @@ int fmi3_xml_handle_Variable(fmi3_xml_parser_context_t* context, const char* dat
     if (!data) {
         fmi3_xml_model_description_t* md = context->modelDescription;
         fmi3_xml_variable_t* variable;
-        fmi3_xml_variable_t dummyV;
-        const char* description = 0;
-        jm_named_ptr named, *pnamed;
-        jm_vector(char)* bufName  = fmi3_xml_reserve_parse_buffer(context, 1, 100);
-        jm_vector(char)* bufDescr = fmi3_xml_reserve_parse_buffer(context, 2, 100);
+        const char* description = NULL;
+        jm_vector(char)* bufName = fmi3_xml_reserve_parse_buffer(context, 1, 100);
+        jm_vector(char)* bufDesc = fmi3_xml_reserve_parse_buffer(context, 2, 100);
         unsigned int vr;
 
-        if(!bufName || !bufDescr) return -1;
+        if(!bufName || !bufDesc) return -1;
 
         /* Get vr, name and description */
-        if (fmi3_xml_set_attr_uint32(context, elm_id, fmi_attr_id_valueReference, 1, &vr, 0))         return -1;
-        if (fmi3_xml_set_attr_string(context, fmi3_xml_elmID_Variable, fmi_attr_id_name, 1, bufName)) return -1;
-        if (fmi3_xml_set_attr_string(context, elm_id, fmi_attr_id_description, 0, bufDescr))          return -1;
+        if (fmi3_xml_set_attr_uint32(context, elm_id, fmi_attr_id_valueReference, 1, &vr, 0))  return -1;
+        if (fmi3_xml_set_attr_string(context, elm_id, fmi_attr_id_name,           1, bufName)) return -1;
+        if (fmi3_xml_set_attr_string(context, elm_id, fmi_attr_id_description,    0, bufDesc)) return -1;
 
         /* Return early if undefined variable */
         if (context->skipOneVariableFlag) {
             jm_log_error(context->callbacks, module, "Ignoring variable with undefined vr '%s'", jm_vector_get_itemp(char)(bufName,0));
             return 0;
         }
 
-        /* Get description if exists */
-        if (jm_vector_get_size(char)(bufDescr)) {
-            description = jm_string_set_put(&md->descriptions, jm_vector_get_itemp(char)(bufDescr,0));
+        if (jm_vector_get_size(char)(bufDesc)) {
+            /* Add the description to the model-wide set and retrieve the pointer */
+            description = jm_string_set_put(&md->descriptions, jm_vector_get_itemp(char)(bufDesc, 0));
         }
 
-        /* Add Variable ptr to modelDescription obj */
-        named.ptr = 0;
-        named.name = 0;
-        pnamed = jm_vector_push_back(jm_named_ptr)(&md->variablesByName, named);
-        if (!pnamed) {
-            fmi3_xml_parse_fatal(context, "Could not allocate memory");
-            return -1;
+        // Create the alias and set name at same time:
+        const char* name;
+        if (jm_vector_get_size(char)(bufName) == 0) {  // FIXME: Empty string has been resized to 0
```

Comment:
It's resized to 0 as part of `fmi3_xml_set_attr_string`, which is called from everywhere. I'm not fixing that now